cmd/buildkitd/config/gcpolicy.go

```diff
@@ -104,3 +104,20 @@ func stripQuotes(s string) string {
 	}
 	return s
 }
+
+func DetectDefaultGCCap() DiskSpace {
+	return DiskSpace{Percentage: DiskSpacePercentage}
+}
+
+func (d DiskSpace) AsBytes(root string) int64 {
+	if d.Bytes != 0 {
+		return d.Bytes
+	}
+	if d.Percentage == 0 {
+		return 0
+	}
+
+	avail := getDiskSize(root) * d.Percentage / 100
```

Comment:
When `getDiskSize` returns `defaultCap` (2GB) this would take 10% of that what is not correct.

Reply:
nasty bug! Thanks for catching this, I'd missed that in my refactoring. Opted to just have `getDiskSize` do just that, and error when it's unable to do that.